File: resource-registry/src/main/java/nl/haploid/octowight/registry/data/ResourceRoot.java
package nl.haploid.octowight.registry.data;

public class ResourceRoot {

	private Long resourceId;

	private String resourceType;

	private Long atomId;

	private String atomOrigin;

	private String atomType;

	private Long version;

	public String key() {
		return String.format("%s:%s/%s->%s", getAtomOrigin(), getAtomType(), getAtomId(), getResourceType());
	}

	public Long getResourceId() {
		return resourceId;
	}

	public void setResourceId(final Long resourceId) {
		this.resourceId = resourceId;
	}

	public String getResourceType() {
		return resourceType;
	}

	public void setResourceType(final String resourceType) {
		this.resourceType = resourceType;
	}

	public Long getAtomId() {
		return atomId;
	}

	public void setAtomId(final Long atomId) {
		this.atomId = atomId;
	}

	public String getAtomOrigin() {
		return atomOrigin;
	}

	public void setAtomOrigin(final String atomOrigin) {
		this.atomOrigin = atomOrigin;
	}

	public String getAtomType() {
		return atomType;
	}

	public void setAtomType(final String atomType) {
		this.atomType = atomType;
	}

	public Long getVersion() {
		return version;
	}

	public void setVersion(final Long version) {
		this.version = version;
	}
}

File: resource-registry/src/main/scala/nl/haploid/octowight/registry/data/ResourceRoot.scala
package nl.haploid.octowight.registry.data

import java.lang

import nl.haploid.octowight.registry.repository.ResourceRootDmo

object ResourceRoot {

  def apply(root: Atom, resourceCollection: String) =
    new ResourceRoot(resourceId = null, resourceCollection = resourceCollection, root = root, version = null)

  def apply(resourceRootDmo: ResourceRootDmo) = {
    val atom = new Atom(resourceRootDmo.root.id, resourceRootDmo.root.origin, resourceRootDmo.root.category)
    new ResourceRoot(
      root = atom,
      resourceId = resourceRootDmo.resourceId,
      resourceCollection = resourceRootDmo.resourceCollection,
      tombstone = resourceRootDmo.tombstone,
      version = resourceRootDmo.version)
  }
}

case class ResourceRoot
(
  resourceId: lang.Long,
  resourceCollection: String,
  root: Atom,
  tombstone: Boolean = false,
  version: lang.Long
  ) extends Atomizable {

  def key = s"${root.origin}:${root.category}/${root.id}->$resourceCollection"

  override def toAtom: Atom = new Atom(root.id, root.origin, root.category)
}
